### ISSUE TITLE
`no-immediate-mutation`: Support conditionals

### ISSUE THREAD

**Issue by abrahamguo** [sindresorhus/eslint-plugin-unicorn#3682]
### Description

The rule should detect when immediate mutation happens in a conditional.

Likely only makes sense if the immediate mutation is the only thing in the conditional.

### Examples

```js
// ❌
const array = [1, 2];
if (Math.random()) array.push(3, 4);

// ✅
const array = [1, 2, ...Math.random() ? [3, 4] : []];

// ❌
const object = {foo: 1};
if (Math.random()) object.bar = 2;
else object.baz = 3;

// ✅
const object = {foo: 1, ...Math.random() ? { bar: 2 } : { baz: 3 } };

```

... and so on for the other types handled by this rule.
